library/core/src/iter/traits/iterator.rs

```diff
@@ -1334,6 +1340,67 @@ pub trait Iterator {
     /// assert_eq!(iter.next(), Some(-6));
     /// assert_eq!(iter.next(), None);
     /// ```
+    ///
```

Comment:
These examples are quite long, and I'm not sure they really improve understanding in a way that it's worth adding these. Maybe we can just modify the existing example to show stopping early by returning `None` from the closure.